### DISCUSSION TITLE
CPU only Integration with LAMMPS

### DISCUSSION THREAD

**Discussion by npiroozan** [mir-group/pair_nequip#36] [Q&A] [answered]
Hello,

I hope you are doing well.  Regarding setup of the framework, all steps were successfully implemented with the exception of integration with LAMMPS.  The instructions detailed in colab work well for the case of an NVIDIA GPU, however I am trying to do this for an x86 CPU only system (single node).  The lack of CUDA is giving a series of errors regarding the CMAKE configuration ahead of LAMMPS build.  If possible, could you please detail how one would do this for my case?  Apologies if this is a naive question :).

Thank you very much.

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi @npiroozan ,

Are you using `pair_nequip` or `pair_allegro`?

Regardless, one node CPU only should be very simple. Just download CPU-only `libtorch` from PyTorch, don't build with Kokkos (in the case of `pair_allegro`), and (again in the case of `pair_allegro`) build with `DPKG_OPENMP=ON`.  They key is to not use `libtorch` from your `conda` PyTorch installation if that installation was done with CUDA support.

**Reply by npiroozan** [reply]
Hello,

My intention is to use pair_style Allegro.

Thank you for the instructions, I am using torch 1.11 with cuda support and will then download the C++ torchlib from PyTorch (cpu only) and source those files.
